scripts/create_or_validate_md5s.py

```diff
@@ -0,0 +1,159 @@
+    """
+    b = get_batch(f'{mode} md5 checksums for files in {gs_dir}')
+
+    files = get_blobs_in_directory(gs_dir, billing_project, storage_client)
+    for filepath in files:
+        if filepath.endswith('.md5') or filepath.endswith(skip_filetypes):
+            continue
+        if f'{filepath}.md5' in files and not force_recreate:
+            if mode == 'validate':
+                print('Validating md5 for', filepath)
+                job = b.new_job(f'Validate md5 checksum: {os.path.basename(filepath)}')
+                validate_md5(job, filepath, billing_project, driver_image)
+            else:
+                print(f'{filepath}.md5 already exists, skipping')
+            continue
+        
+        if mode != 'create':
+            continue
+        print('Creating md5 for', filepath)
+        job = b.new_job(f'Create {os.path.basename(filepath)}.md5')
+        create_md5(job, filepath, billing_project, driver_image)
+
+    b.run(wait=False)
+
+
+@click.command()
+@click.option('--billing-project', '-b', default=None)
+@click.option('--skip-filetypes', '-s', default=('.crai', '.tbi'), multiple=True)
+@click.option('--validate-existing-only', '-v', is_flag=True, default=False, help='Validate existing md5s, do not create new ones.')
+@click.option('--create-only', '-c', is_flag=True, default=False, help='Create new md5s, do not validate existing ones.')
+@click.option('--force-recreate-existing', '-f', is_flag=True, default=False, help='Re-create md5s even if they already exist.')
+@click.argument('gs_dir')
+def main(
+    billing_project: str | None,
+    skip_filetypes: tuple[str, str],
+    validate_existing_only: bool,
+    create_only: bool,
+    force_recreate_existing: bool,
+    gs_dir: str
+):
+    """
+    Scans the directory for files and creates md5 checksums for them, OR validates existing md5s.
+    params:
+        gs_dir:                   The GCP bucket path / directory to scan for files.
+        skip_filetypes:           File extensions to skip.
+        *validate_existing_only:  Validate existing md5s, do not create new ones.
+        *create_only:             Create new md5s, do not validate existing ones.
+        force_recreate_existing:  Re-create md5s even if they already exist.
+        billing_project:          The GCP project to bill for the operations.
+
+        *Only one of validate_existing or create_only can be set.
+    """
+    if not gs_dir.startswith('gs://'):
+        raise ValueError(f'Expected GS directory, got: {gs_dir}')
+
+    if not billing_project:
+        billing_project = config_retrieve(['workflow', 'dataset_gcp_project'])
+    driver_image = config_retrieve(['workflow', 'driver_image'])
+
+    if validate_existing_only and create_only:
+        raise ValueError('Only one of --validate-existing-only or --create-only can be set.')
+
+    mode = 'validate' if validate_existing_only else 'create'
```

Comment:
Directly ask the user for the mode as a click argument with only two choices (`create` / `validate`), rather than inferring it through a set of command line options that are unbound.